module.go

```diff
@@ -131,6 +135,13 @@ func (m *module) build(app *App, root *dig.Container) {
 	}
 }
 
+func (m *module) getTrace() (trace fxreflect.Stack) {
+	if m.parent != nil {
+		trace = m.parent.getTrace()
+	}
+	return append(trace, m.location)
```

Comment:
Usually stack traces start from the latest frame and work its way down. As it is, the order of module trace is reverse of stack trace. I think this might be confusing to users. Also, I think it might be helpful if we can somehow include module's name to the module trace.

Reply:
Makes sense - I updated to have the reverse order. This is probably less efficient due to the repeated `append([]string{location}, locations...)`, but I agree with you about it being more correct. I thought about keeping the order and doing a slice reversal at the end, but thought that might be overkill.

Reply:
@tchung1118 and I talked offline a little.

I switched the PR to cache traces for modules when creating them, rather than re-chaining the frames for every provide/decorate/supply/replace. I think this should be faster, especially for applications with lots of constructors/decorators.